TS compilation error: Cannot find name 'GenericObject'

**Describe the bug**
Using redoc in a project with `skipLibCheck: false` , I get the following error:

```
.yarn/__virtual__/redoc-virtual-0d2672ebad/1/build/node/cache/redoc-npm-2.0.0-f00948f5d8-8.zip/node_modules/redoc/typings/services/OpenAPIParser.d.ts:15:20 - error TS2304: Cannot find name 'GenericObject'.

15     validate(spec: GenericObject): void;
                      ~~~~~~~~~~~~~
```

It presumably refers to the definition here: 
https://github.com/Redocly/redoc/blob/16f542bbfe856e5ae5eb40636ac79421fa2f397b/custom.d.ts#L27

but as things stand, either I'm holding it wrong or the shipped typescript stubs are faulty.

**Expected behavior**
No errors on typescript compilation

**Additional context**
Typescript 4.6.3, redoc 2.0.0

--- a/custom.d.ts
+++ b/custom.d.ts
@@ -23,5 +23,3 @@ declare var reactHotLoaderGlobal: any;
 interface Element {
   scrollIntoViewIfNeeded(centerIfNeeded?: boolean): void;
 }
-
-type GenericObject = Record<string, any>;

--- a/src/components/DropdownOrLabel/DropdownOrLabel.tsx
+++ b/src/components/DropdownOrLabel/DropdownOrLabel.tsx
@@ -4,7 +4,7 @@ import { StyledComponent } from 'styled-components';
 import { DropdownProps, MimeLabel, SimpleDropdown } from '../../common-elements/Dropdown';
 
 export interface DropdownOrLabelProps extends DropdownProps {
-  Label?: StyledComponent<any, any, GenericObject, never>;
+  Label?: StyledComponent<any, any, Record<string, any>, never>;
   Dropdown?: StyledComponent<
     React.NamedExoticComponent<DropdownProps>,
     any,

--- a/src/services/OpenAPIParser.ts
+++ b/src/services/OpenAPIParser.ts
@@ -41,7 +41,7 @@ export class OpenAPIParser {
     }
   }
 
-  validate(spec: GenericObject): void {
+  validate(spec: Record<string, any>): void {
     if (spec.openapi === undefined) {
       throw new Error('Document must be valid OpenAPI 3.0.0 definition');
     }
@@ -153,7 +153,7 @@ export class OpenAPIParser {
     } else {
       // small optimization
       return {
-        ...(resolved as GenericObject),
+        ...(resolved as object),
         ...rest,
       } as T;
     }

--- a/src/utils/object.ts
+++ b/src/utils/object.ts
@@ -1,4 +1,4 @@
-export function objectHas(object: GenericObject, path: string | Array<string>): boolean {
+export function objectHas(object: object, path: string | Array<string>): boolean {
   let _path = <Array<string>>path;
 
   if (typeof path === 'string') {
@@ -12,7 +12,7 @@ export function objectHas(object: GenericObject, path: string | Array<string>): 
   });
 }
 
-export function objectSet(object: GenericObject, path: string | Array<string>, value: any): void {
+export function objectSet(object: object, path: string | Array<string>, value: any): void {
   let _path = <Array<string>>path;
 
   if (typeof path === 'string') {

